pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -159,6 +159,10 @@ func (p *ecrProvider) Provide(image string) credentialprovider.DockerConfig {
 	// (see https://github.com/kubernetes/kubernetes/issues/92162)
 	once.Do(func() {
 		isEC2 = p.isEC2()
+
+		if isEC2 && credentialprovider.AreLegacyCloudCredentialProvidersDisabled() {
```

Comment:
also need to actually add this to disable this provider after the `if !isEC2 {` block below

```go
if credentialprovider.AreLegacyCloudCredentialProvidersDisabled() {
  return credentialprovider.DockerConfig{}
}
```

Reply:
Oh of course, good catch. Fixed.